test/metabase/driver/h2_test.clj

```diff
@@ -415,3 +416,49 @@
            (sql-jdbc.actions/maybe-parse-sql-error
             :h2 actions.error/violate-foreign-key-constraint {:id 1} :row/update
             "Referential integrity constraint violation: \"USER_GROUP-ID_GROUP_-159406530: PUBLIC.\"\"USER\"\" FOREIGN KEY(\"\"GROUP-ID\"\") REFERENCES PUBLIC.\"\"GROUP\"\"(ID) (CAST(999 AS BIGINT))\"; SQL statement:\nINSERT INTO \"PUBLIC\".\"USER\" (\"NAME\", \"GROUP-ID\") VALUES (CAST(? AS VARCHAR), CAST(? AS INTEGER)) [23506-214]")))))
+
+(deftest column-attributes-test
+  (mt/test-driver :h2
+    (mt/with-empty-db
+      (let [spec (mdb/spec :h2 (:details (mt/db)))
+            describe
+            (fn [column-defs]
+              (jdbc/execute! spec ["DROP TABLE IF EXISTS TEST_TABLE"])
+              (jdbc/execute! spec [(->> (map #(format "col%d %s" %1 %2) (range) column-defs)
+                                        (str/join ",")
+                                        (format "CREATE TABLE TEST_TABLE (%s)"))])
+              (->> (driver/describe-table :h2 (mt/db) {:name "TEST_TABLE"})
+                   :fields
+                   (sort-by :database-position)
+                   (mapv #(-> (select-keys % [:database-is-generated :database-is-nullable :database-default])
+                              ;; shorthand
+                              (set/rename-keys {:database-is-generated :g
+                                                :database-is-nullable  :n
+                                                :database-default      :d})))))
+            do-test
+            (fn [& test-cases]
+              (let [test-cases' (partition 2 test-cases)
+                    results (describe (map first test-cases'))]
+                (is (= (count test-cases) (* 2 (count results))))
+                (doseq [[[col-def expected] result] (map vector test-cases' results)]
+                  (testing col-def
+                    (is (= expected result))))))]
+        (do-test
```

Comment:
hmmm is this `clojure.test/are`? :-)

Reply:
I did consider `are`, the plain statement of inputs and outputs is certainly what I'm going for here. However in this case the resource lifetime is for all the inputs (as we create a column per input string) not each input (which would be more convenient for `are`). I could convolute the code to suit `are` but not sure if it would be better tbh 🤷 . Honestly I would leave it as is - but what do you think? Worth rewriting it?

Reply:
ahhh right; I comment just because metabase is full of test helpers which then misreport the line where test failed, consider that my personal pet peeve. 😁 Did not notice one big "describe" running for everything. 🤷‍♂️